### ISSUE TITLE
MediaInfo.Get(inputPath): Object reference not set to an instance of an object

### ISSUE THREAD

**Issue by marchanddesable** [tomaszzmuda/Xabe.FFmpeg#181]
Hi,

When I take a video with my android phone, and try to use the ffmpeg library to convert and resize it, in a .net core projet (in docker/linux), I have the following issue:

If I do: 
MediaInfo.Get(inputPath)

it crashes and the error is:

Object reference not set to an instance of an object
at Xabe.FFmpeg.FFmpeg..ctor()
   at Xabe.FFmpeg.MediaInfo.Get(FileInfo fileInfo)
   at Xabe.FFmpeg.MediaInfo.Get(String filePath)

(inputPath in the path of the .mp4 source file, this file exists.)

A few months ago it worked, but now I can't find any solution.

Thank you.

**Comment by 06needhamt** [comment]
Hi, @marchanddesable would you be able to provide a link to a video which causes this problem so we can test? Also which version of Xabe.FFmpeg are you using the latest is v3.3.0. If you are not using this version, please could you try with the latest version to see if the issue still occurs? If so I would be happy to investigate.

**Comment by Gamingdevil** [comment]
I had the same issue when I deployed my application to Azure. As I recall, it was because ffprobe.exe was not available. Is it possible that you do not have the file present? 
I'm not entirely sure why it needs this, because I don't use any probe functionality, but that fixed it for me.

I also get the impression that MediaInfo.Get is arbitrarily slow sometimes, where it wasn't during my initial tests, but maybe that's just me.

**Comment by marchanddesable** [comment]
Hi,

var mediaInfo = await MediaInfo.Get(pathOfTheFile)
The issue was a wrong path, and the error info did not help...

Another issue:

var vidConv = Conversion.New()
                        .AddParameter(...)
                        .AddParameter(...)
                        .AddParameter(...)
                       .SetOutput(destPath);

var result = await vidConv.Start();

Sometimes, the Start function takes a VERY long time to finish, I'm not even sure it finishes, because I did not wait more than 5 minutes (the video encoded was less than 20 seconds)...

I spent 2 days on this issue, sometimes it works but sometime it just stuck and wait forever...

When I have added the progress calback I have discovered that the process was finished but the process was blocked somewhere and could not return...

So here is what I have done, as workaround:

The OnProgress callback works, so I know when the process is finished (100%), but the await ..Start() sometimes get stuck and never return, so my Idea was to raise a cancell event when the end of the task was detected in the callback.

CancellationTokenSource clt = new CancellationTokenSource();
CancellationToken tok = clt.Token;
bool canCancell = false;
bool succeeded = false;

vidConv.OnProgress += (sndr, args) =>
{
      if (args.Percent == 100)
       {
           if (!canCancell)
           {
               canCancell = true;
                                                    
                clt.CancelAfter(8000); //I send it 8s after the end being detected.
          }
       }
 };


 var tsk = vidConv.Start(); //start the conversion process

 try
 {
      tsk.Wait(tok);
 }
 catch (Exception ex)
 {
     //The cancellation error is catched
     succeeded = true;
 }

//if it finished properly without using the cancellation token, just take the result of the task.
 if (!succeeded)
 {
      if (tsk?.Result != null)
         succeeded = tsk.Result.Success;
}

if (succeeded)
{
  //here, I assume the encoding has succeeded.
}

That's the only solution I have found, I still don't know why the normal "await vidConv.Start()" doesn't returns for some videos.

Note: I use a samsung galaxy S7 edge, the feature is: you take a video with the phone, you push the send button and a few second later the process starts (in a docker container with linux in kubernetes).
